File: psp/basicGraphql/graphqlServer/GraphqlBasicServer/src/main/java/com/example/graphqlbasicserver/domain/services/ServicesShop.java
package com.example.graphqlbasicserver.domain.services;


import com.example.graphqlbasicserver.data.GameEntityRepository;
import com.example.graphqlbasicserver.data.ShopEntityRepository;
import com.example.graphqlbasicserver.data.model.mapper.ShopMapper;
import com.example.graphqlbasicserver.domain.model.Shop;
import lombok.extern.log4j.Log4j2;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
@Log4j2
public class ServicesShop {

    private final ShopEntityRepository shopR;

    private final GameEntityRepository gameR;

    private final ShopMapper shopMapper;

    public ServicesShop(ShopEntityRepository shopR, GameEntityRepository gameR, ShopMapper shopMapper) {
        this.shopR = shopR;
        this.gameR = gameR;
        this.shopMapper = shopMapper;
    }

    public List<Shop> getAll() {
        return shopR.findAll()
                .stream()
                .map(shopMapper::toShop)
                .toList();
    }

    public Shop get(int id) {
        return shopR.findById(id).map(shopMapper::toShop).orElse(null);
    }

    public Shop add(Shop shop) {
        return shopMapper.toShop(shopR.save(shopMapper.toShopEntity(shop)));
    }

    public Shop update(Shop shop) {
        return shopMapper.toShop(shopR.save(shopMapper.toShopEntity(shop)));
    }

    public boolean delete(int id) {
        try {
            gameR.deleteFromShop(id);
            shopR.deleteById(id);
        } catch (Exception e) {
            return false;
        }
        return true;
    }

    public List<Shop> filterByName(String name) {
        return shopR.findAll()
                .stream()
                .map(shopMapper::toShop)
                .filter(shop -> shop.getName().trim().toLowerCase().contains(name.trim().toLowerCase()))
                .toList();
    }
}

File: psp/basicGraphql/graphqlClient/graphqlClient/src/main/kotlin/org/example/domain/services/ServicesShop.kt
package org.example.domain.services

import org.example.data.dao.DaoShops
import org.example.domain.model.Shop

object ServicesShop {
    private val dS: DaoShops = DaoShops

    fun getAllShops() = dS.getAllShops()
    fun getShopsFilterByName(name: String) = dS.getShopsFilterByName(name)
    fun getShop(id: Int) = dS.getShop(id)
    fun addShop(name: String) = dS.addShop(name)
    fun updateShop(shop: Shop) = dS.updateShop(shop)
    fun deleteShop(id: Int) = dS.deleteShop(id)
}